### ISSUE TITLE
Search for symbols does not consider settings.exclude after window reload.

### ISSUE THREAD

**Issue by wataBottl** [microsoft/vscode#209902]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.88.0
- OS Version: Windows_NT x64 10.0.22631

Steps to Reproduce:

1. Open workspace folder/reload window
2. Search for function name in quick open bar
3. Excluded folder (under `**/build`) appears -> not ok
![image](https://github.com/microsoft/vscode/assets/93131435/010a7ae7-4931-48a4-aa03-c91d8757b4bc)

Opening the declaration file in `build`, closing it and searching for the same symbol again seems to display only the correct file in the `src` folder

**Comment by andreamah** [comment]
I don't know whether I'd expect `search.exclude` to apply in this case. It's more of workspace and file search setting. @mjbvz @TylerLeonhardt do you have any opinions on this?

**Comment by jrieken** [comment]
> It's more of workspace and file search setting.

Agree. We have been saying that it's up to extensions to honour this flag or to offer the right project/extension configurations for this. Doing the filtering on our side (e.g inside the UI) is wasteful because resources to compute the results have already been spend
